diff -up gedit-3.8.3/configure.ac.disable-python gedit-3.8.3/configure.ac
--- gedit-3.8.3/configure.ac.disable-python	2013-07-17 20:41:04.210713683 -0400
+++ gedit-3.8.3/configure.ac	2013-07-17 20:41:36.763967930 -0400
@@ -306,7 +306,7 @@ if test "x$enable_python" = "xauto"; the
 fi
 
 if test "x$enable_python" = "xyes"; then
-	AM_PATH_PYTHON(3.2.3)
+	AM_PATH_PYTHON
 	PKG_CHECK_MODULES(PYTHON, [pygobject-3.0 >= $PYGOBJECT_REQUIRED])
 
 	pyoverridesdir=`$PYTHON -c "import gi; print(gi._overridesdir)"`
diff -up gedit-3.8.3/gedit/gedit-plugins-engine.c.disable-python gedit-3.8.3/gedit/gedit-plugins-engine.c
--- gedit-3.8.3/gedit/gedit-plugins-engine.c.disable-python	2013-03-12 02:35:32.000000000 -0400
+++ gedit-3.8.3/gedit/gedit-plugins-engine.c	2013-07-17 20:41:04.210713683 -0400
@@ -66,7 +66,7 @@ gedit_plugins_engine_init (GeditPluginsE
 	                                            GEDIT_TYPE_PLUGINS_ENGINE,
 	                                            GeditPluginsEnginePrivate);
 
-	peas_engine_enable_loader (PEAS_ENGINE (engine), "python3");
+	peas_engine_enable_loader (PEAS_ENGINE (engine), "python");
 
 	engine->priv->plugin_settings = g_settings_new ("org.gnome.gedit.plugins");
 
--- a/plugins/externaltools/externaltools.plugin.desktop.in
+++ b/plugins/externaltools/externaltools.plugin.desktop.in
@@ -1,5 +1,5 @@ 
 [Plugin]
-Loader=python3
+Loader=python
 Module=externaltools
 IAge=3
 _Name=External Tools
--- a/plugins/pythonconsole/pythonconsole.plugin.desktop.in
+++ b/plugins/pythonconsole/pythonconsole.plugin.desktop.in
@@ -1,5 +1,5 @@ 
 [Plugin]
-Loader=python3
+Loader=python
 Module=pythonconsole
 IAge=3
 _Name=Python Console
--- a/plugins/quickopen/quickopen.plugin.desktop.in
+++ b/plugins/quickopen/quickopen.plugin.desktop.in
@@ -1,5 +1,5 @@ 
 [Plugin]
-Loader=python3
+Loader=python
 Module=quickopen
 IAge=3
 _Name=Quick Open
--- a/plugins/snippets/snippets.plugin.desktop.in
+++ b/plugins/snippets/snippets.plugin.desktop.in
@@ -1,5 +1,5 @@ 
 [Plugin]
-Loader=python3
+Loader=python
 Module=snippets
 IAge=3
 _Name=Snippets
--- a/plugins/snippets/snippets/shareddata.py
+++ b/plugins/snippets/snippets/shareddata.py
@@ -20,7 +20,10 @@ import os
 
 from gi.repository import Gtk
 
-class SharedData(object, metaclass=Singleton):
+
+class SharedData(object):
+    __metaclass__ = Singleton
+
     def __init__(self):
         self.dlg = None
         self.dlg_default_size = None
@@ -29,7 +32,7 @@ class SharedData(object, metaclass=Singl
 
     def register_controller(self, view, controller):
         self.controller_registry[view] = controller
-    
+
     def unregister_controller(self, view, controller):
         if self.controller_registry[view] == controller:
             del self.controller_registry[view]
@@ -72,7 +75,8 @@ class SharedData(object, metaclass=Singl
             self.dlg.connect('destroy', self.manager_destroyed)
 
             if self.dlg_default_size:
-                self.dlg.set_default_size(self.dlg_default_size[0], self.dlg_default_size[1])
+                self.dlg.set_default_size(self.dlg_default_size[0],
+                                          self.dlg_default_size[1])
 
         self.dlg.set_transient_for(window)
         self.dlg.present()

