diff -up gedit-3.14.3/gedit/gedit-recent.c.jp-search gedit-3.14.3/gedit/gedit-recent.c
--- gedit-3.14.3/gedit/gedit-recent.c.jp-search	2015-08-18 10:11:59.995992219 -0400
+++ gedit-3.14.3/gedit/gedit-recent.c	2015-08-18 10:12:23.350315762 -0400
@@ -250,7 +250,7 @@ gedit_recent_get_items (GeditRecentConfi
 			{
 				gchar *uri_lower;
 
-				uri_lower = g_utf8_strdown (gtk_recent_info_get_uri (info), -1);
+				uri_lower = g_utf8_strdown (gtk_recent_info_get_uri_display (info), -1);
 
 				if (strstr (uri_lower, config->substring_filter) == NULL)
 				{
